test/e2e/specs/editor/various/inner-blocks-templates.spec.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Inner blocks templates', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activatePlugin(
+			'gutenberg-test-inner-blocks-templates'
+		);
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin(
+			'gutenberg-test-inner-blocks-templates'
+		);
+	} );
+
+	test( 'applying block templates asynchronously does not create a persistent change in the editor', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'test/test-inner-blocks-async-template',
+		} );
+
+		const blockWithTemplateContent = page.locator(
+			'role=document[name="Block: Test Inner Blocks Async Template"i] >> text=OneTwo'
+		);
+
+		// The block template content appears asynchronously, so wait for it.
+		await blockWithTemplateContent.waitFor();
```

Comment:
Nit: I think this would be the same as the assertion below. I personally prefer more explicit assertions.

```js
await expect( blockWithTemplateContent ).toBeVisible();
```